source/Patches/Roles/RoleSingleton.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Linq;
+using Reactor;
+using static TownOfUs.Roles.Role;
+
+namespace TownOfUs.Roles
+{
+    public static class RoleSingleton<T> where T : Role
+    {
+        private static T _instance;
+        public static T Instance => _instance ??= RoleSingleton.OfType<T>().Single();
+
+        internal static void LoadSingletons()
+        {
+            foreach (var type in typeof(TownOfUs).Assembly.GetTypes())
+            {
+                if (!type.IsSubclassOf(typeof(Role))) continue; 
+                AddSingleton(Activator.CreateInstance(type) as Role);
```

Comment:
Does this Activator.CreateInstance call cause the role object to be put into the player-role dictionary?

Reply:
No. Instead, it will call the parameter-less constructor since we have no object (that is used for parameter) included.

Reply:
The `RoleDictionary` stuff already makes me nervous because it's action at a distance. I'm really nervous about this "sometimes it contains a player, sometimes it only contains some text and configuration stuff".

My initial reaction is to put another level of indirection in the way and have an intermediate class containing that stuff the roles implement. But I suspect that doesn't actually end up any cleaner, it just hides the problem.